modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/util/Commons.java

```diff
@@ -598,12 +600,12 @@ public static int nextPowerOf2(int v) {
      * @param elems Elems.
      * @return The longest possible prefix of {@code seq}.
      */
-    public static <T> List<T> maxPrefix(List<T> seq, Collection<T> elems) {
-        List<T> res = new ArrayList<>();
+    public static List<Integer> maxPrefix(List<Integer> seq, Collection<Integer> elems) {
+        List<Integer> res = new IntArrayList();
 
-        Set<T> elems0 = new HashSet<>(elems);
+        Set<Integer> elems0 = new IntOpenHashSet(elems);
 
-        for (T e : seq) {
+        for (Integer e : seq) {
             if (!elems0.remove(e)) {
                 break;
             }
```

Comment:
seems it could be rewritten at all without Integer

Reply:
the comment relate to all amended places in the PR

Reply:
Done